### PR TITLE
[CHK-3119] feat: map `WISP_REDIRECT` client id when creating transactions to `CHECKOUT_CART`

### PR DESCRIPTION
**Pull request by giovanniberti**
#### Motivation and Context

This PR includes the groundwork from pagopa/pagopa-ecommerce-commons#214
The returned field `clientId` is mapped to `CHECKOUT_CART` in order to make the new `WISP` client backward compatible with the current API

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Depends on pagopa/pagopa-ecommerce-commons#214

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -559,8 +559,8 @@
                 <configuration>
                     <skipCheckoutIfExists>false</skipCheckoutIfExists>
                     <connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
-                    <scmVersionType>tag</scmVersionType>
-                    <scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+                    <scmVersionType>branch</scmVersionType>
+                    <scmVersion>CHK-3104-wisp-dismantling-client-id</scmVersion>
```
Comment (giovanniberti) [inline pom.xml:563]: TODO fix after depending PR merge

File: src/main/java/it/pagopa/transactions/services/v2_1/TransactionsService.java [hunk 2]
```diff
@@ -160,19 +160,22 @@ private Mono<NewTransactionResponseDto> projectActivatedEvent(
...
-                                                                  String clientId
+                                                                  Transaction.ClientId clientId
     ) {
         return Optional.ofNullable(clientId).filter(Objects::nonNull)
                 .map(
                         value -> {
                             try {
-                                return NewTransactionResponseDto.ClientIdEnum.fromValue(value);
+                                return switch (clientId) {
+                                    case WISP -> NewTransactionResponseDto.ClientIdEnum.CHECKOUT_CART;
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/services/v2_1/TransactionsService.java:176]:
```suggestion
                                    case WISP_REDIRECT -> NewTransactionResponseDto.ClientIdEnum.CHECKOUT_CART;
```

Comment (giovanniberti) [reply src/main/java/it/pagopa/transactions/services/v2_1/TransactionsService.java]: Fixed in f879b87f

File: api-spec/v2.1/transactions-api.yaml [hunk 3]
```diff
@@ -278,6 +278,7 @@ components:
         - IO
         - CHECKOUT
         - CHECKOUT_CART
+        - WISP
```
Comment (petretiandrea) [inline api-spec/v2.1/transactions-api.yaml:281]: In order to avoid breaking changes we had decided not to add clientId or am I wrong?

Comment (giovanniberti) [reply api-spec/v2.1/transactions-api.yaml:281]: This is used only in `POST /transactions`, so it's not breaking
It's still needed as a change to support the use case of activating a transaction without calling verify on it beforehand

**Review by petretiandrea — APPROVED** [review]

**Pull request merged**